draft-bar-cfrg-spake2plus.md

```diff
@@ -251,31 +251,37 @@ x <- [0, p-1]
 X = x*P + w0*M
 ~~~
 
-Upon receipt of X, the verifier checks the received element for group membership
+The verifier, when it receives X, checks the received element for group membership
 and aborts if X is not in the large prime-order subgroup of G; see {{security}}
 for details. The verifier then selects y uniformly at random from the integers
 in [0, p-1], computes the public share shareV=Y and transmits it to the prover.
-Upon receipt of Y, the prover checks the received element for group membership
-and aborts if Y is not in the large prime-order subgroup of G.
+It then computes Z and V, which will be the common shared values:
 
 ~~~
-y <- [0, p-1]
-Y = y*P + w0*N
-~~~
+Verifier_receives_X:
```

Comment:
I wonder if it would be more clear if we were to write each step in terms of explicit functions, and then show how they're called? 

For example, these might be the functions:

```
def ProverInit:
   // Compute Prover key share
   x <- [0, p-1]
   X = x*P + w0*M
   return (x, X)

def VerifierFinish(X):
   if not_in_subgroup(X):
      raise "invalid input"
   
   // Compute Verifier key share
   y <- [0, p-1]
   Y = y*P + w0*N
   
   // Compute shared values
   Z = h*y*(X - w0*M)
   V = h*y*L

   return (Z, V)

def ProverFinish(x, Y):
   if not_in_subgroup(Y):
      raise "invalid input"

   // Compute shared values
   Z = h*x*(Y - w0*N)
   V = h*w1*(Y - w0*N)

   return (Y, Z, V)
```

And then this might be how we describe the protocol:

```
Prover:
   (x, X) = ProverInit()
   Transmit(X)
   Y = Receive()
   (Z, V) = ProverFinish(x, Y)
   // ... write how to use (Z, V)?

Verifier:
   X = Receive()
   (Y, Z, V) = VerifierFinish(X)
   Transmit(Y)
   // ... write how to use (Z, V)?
```

And I might just include this in an appendix, [similar to RFC8446](https://datatracker.ietf.org/doc/html/rfc8446#appendix-A). What do you think?

Reply:
I implemented this as a separate PR, #33, so we have both versions to look at.